addons/osfstorage/models.py

```diff
@@ -560,3 +561,20 @@ def create_waterbutler_log(self, auth, action, metadata):
             auth=auth,
             params=params
         )
+
+    @property
+    def storage_usage(self):
+        storage_usage = cache.get('storage_usage')
+
+        if storage_usage:
+            return storage_usage
+
+        file_versions = OsfStorageFile.objects.filter(provider='osfstorage',
+                                                   target_object_id=self.owner_id,
+                                                   versions__isnull=False).values_list('versions',
+                                                                                       flat=True)
+        # An empty queryset will produce a None instead of 0
+        sum = FileVersion.objects.filter(id__in=file_versions).aggregate(sum=models.Sum('size'))['sum'] or 0
+
+        cache.set('storage_usage', sum, 60)
```

Comment:
You'll need to cache the values per-node. Otherwise, we'd be return the most recently-computed value for *every node*. To do this, come up with a unique cache key for each node, e.g. `storage_usage:123` (where 123 is the PK of the node).

Reply:
Also, a nit: `60` is a magic value. Better to set it as a constant.


```python
cache.set(cache_key, value, self.STORAGE_USAGE_CACHE_TIMEOUT)
```